.github/workflows/check_authors.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+# Only check the first 10 email addresses found in the PR
+EMAILS=$(git log $1 --pretty=format:"%ae" | sort | uniq | head -n 10)
+
+for EMAIL in $EMAILS ; do
+  # Check if the email exists in the AUTHORS file
+  if grep -Fq "$EMAIL" AUTHORS; then
+    echo "Author's email is in the AUTHORS file."
+  else
+    # Post a comment on the PR
+    COMMENT="Hello! Welcome and thank you for your contribution. It appears that you might be a new contributor to our project. To acknowledge your work appropriately, we kindly ask you to add your email to the AUTHORS file. This helps us maintain a record of all our valuable contributors. Thanks again for your involvement!"
+    PAYLOAD=$(jq -n --arg body "$COMMENT" '{body: $body}')
+    COMMENTS_URL=$(jq -r .pull_request.comments_url "$GITHUB_EVENT_PATH")
+
+    curl -s -S -H "Authorization: Bearer $GITHUB_TOKEN" --header "Content-Type: application/json" --data "$PAYLOAD" "$COMMENTS_URL"
+  fi
+done
```

Comment:
Wondering whether we can allow names without emails.  I already get a lot of spam at my main email address that I'm reluctant to put it in a file that is easy to crawl.

Reply:
sure.. that's not hard to implement